Library/Homebrew/os/mac/keg.rb

```diff
@@ -8,6 +8,7 @@ def change_dylib_id(id, file)
     @require_relocation = true
     odebug "Changing dylib ID of #{file}\n  from #{file.dylib_id}\n    to #{id}"
     MachO::Tools.change_dylib_id(file, id, strict: false)
+    apply_ad_hoc_signature(file)
```

Comment:
I think it would be nice to keep this logic in ruby-macho with `MachO::Tools.codesign` or similar which would allow replacing it with a native (i.e. not shelling out to `codesign`) version at some point in the future. See also: https://github.com/Homebrew/brew/pull/9040#discussion_r517174497